lib/util/jsonata.ts

```diff
@@ -0,0 +1,18 @@
+import jsonata from 'jsonata';
+import * as memCache from './cache/memory';
+
+export function getExpression(input: string): jsonata.Expression | Error {
+  const cacheKey = `jsonata:${input}`;
```

Comment:
I'm on my phone, do we hash the value afterwards? 

This transformations can become quite big.

Reply:
memCache is a simple `Record<string, any>`. We need a way to avoid repeated compilations if they're expensive, particularly for packageRules.matchJsonata which will be evaluated many times

Reply:
I'm not questioning that caching is useful here, but rather I'm concerned of the potentional key sizes

Reply:
Gotcha, I guess we could hash it

Reply:
Now hashed